### PR TITLE
Resources/edit module

### PR DESCRIPTION
**Pull request by danielkao14**
User is able to edit the title and body of a module
**Changes:** 

- api call to editTextField (takes in text, id of module, which field to change)
- title, body of module changed to MUI Textfields
- additional states/functions in module.js component to store state of editable texts

### CODE REVIEW

**Review by erinkwon01 — APPROVED** [review]
Good job Daniel "I'm not giving you my middle name" Kao!

**Review by galenheuer — APPROVED** [review]

File: client/src/components/NewModulePopup.jsx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+    uploadTask.on(
+      'state_changed',
+      (snapshot) => {
+        const p = Math.round(
+          (snapshot.bytesTransferred / snapshot.totalBytes) * 100,
+        );
+
+        // update progress
+        // setPercent(p);
```
Comment (galenheuer) [inline client/src/components/NewModulePopup.jsx:64]: file upload seems to be working perfectly so you can remove the setPercent line and the console.log!

File: client/src/components/NewModulePopup.jsx [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+        targetServiceAreas.push(SA);
+      }
+    });
+    return targetServiceAreas;
+  };
+
+  const submitForm = async (e) => { // adds a module to the root module page
+    e.preventDefault();
+    const selectedSAs = selectedServiceAreas();
+    console.log(selectedSAs);
```
Comment (galenheuer) [inline client/src/components/NewModulePopup.jsx:85]: can remove

File: client/src/pages/ExpandedModule.jsx
Comment (galenheuer) [inline client/src/pages/ExpandedModule.jsx:1]: one thing i noticed about the edit module functionality is that when you edit the module title in the MUI textfield, the text in the field stays in the updated state upon clicking 'save' but the actual title of the module doesn't change. to be honest I couldn't figure out where in the code that would need to change though lol

**Review by galenheuer — APPROVED** [review]

**Review by xu-jerry — APPROVED** [review]
Looks good! Thanks for reviewers as well!

**Pull request merged**